internal/address/encode.go

```diff
@@ -22,9 +22,10 @@ import (
 
 // EncodeByProtocol takes an address as `[]byte` then selects the relevant encoding method to encode it as string.
 func EncodeByProtocol(in []byte, protocol string) (encoded, encodingType string, err error) {
+	encodingType = encoding.TypeHex0XPrefix
```

Comment:
I would suggest moving this inside the case statement for each case. It then is clearer for someone looking at this in the future that `encodingType` can change based on protocol. Although its more code, it is clearer. Always best to be clear.